src/other/index.jl

```diff
@@ -214,8 +214,28 @@ end
 @inline Base.getindex(x::AbstractIndex, idx::AbstractRange{<:Integer}) =
     getindex(x, collect(Int, idx))
 @inline Base.getindex(x::AbstractIndex, ::Colon) = Base.OneTo(length(x))
-@inline Base.getindex(x::AbstractIndex, notidx::Not) =
+
+@inline Base.getindex(x::AbstractIndex, notidx::Not{<:AbstractVector}) =
```

Comment:
As I have noted already this definition is incorrect in general (e.g. it does not correctly handle boolean indexing).
Therefore I would prefer to narrow it down to 
```suggestion
@inline Base.getindex(x::AbstractIndex, notidx::Union{Not{<:AbstractVector{Symbol}}, Not{<:AbstractVector{<:AbstractString}}}) =
```